### ISSUE TITLE
Make Breadcrumbs Clickable

### ISSUE THREAD

**Issue by slightknack** [zed-industries/zed#5272]
### Description

Sometimes when browsing code, breadcrumbs will contain types and functions you'd like to jump to. For example, given the following breadcrumb:

```
project/src/example.rs  >  impl ThisTrait for CustomType  >  fn display
```

Clicking `impl` should jump to the beginning of the implementation, `ThisTrait` to the definition of that trait, `CustomType` to the definition of that type, and `display` to the beginning of the function you're currently in.

**Comment by Moshyfawn** [comment]
> Would be great if each segment of the path is clickable too, which shows a dropdown and lets you navigate and open other files under each directory, like VS Code does, but this would probably warrant a separate issue.

Someone has created a separate feature request for what you've mentioned here in your post #9996.

**Comment by sirenkovladd** [comment]
Yes, this is still relevant, please leave this issue open, it's actually the only reason why I can't use Zed, otherwise it's the best editor for me

**Comment by OmChillure** [comment]
@esthertrapadoux can you assign this to me?

**Comment by TwistingTwists** [comment]
Hey @OmChillure , I am almost done. Should be up today:).
